git/intermediate_git/working_with_remotes.md

```diff
@@ -0,0 +1,89 @@
+### Dangers and best practices
+
+Let's review the dangers we've addressed so far. I know, I know, it's scary stuff - but we have to be mindful or our coworkers might end up hating our guts! If you look back through this lesson you'll see a common thread. `amend`, `rebase`, `reset`, `push --force` are all especially dangerous when you're collaborating with others. <span id='dangers'>These commands can destroy work your coworkers have created</span>. So keep that in mind. When attempting to rewrite history, always check the dangers of the particular command you're using and follow these best practices for the commands we've covered:
+
+<span id='best-practices'></span>
+
+1.  If working on a team project, make sure rewriting history is safe to do and others know you're doing it.
+2.  Ideally, stick to using these commands only on branches that you're working with by yourself.
+3.  Using the `-f` flag to force something should scare you, and you better have a really good reason for using it.
+4.  Don't push after every single commit, changing published history should be avoided when possible.
+5.  Regarding the specific commands we've covered:
+    1.  For `git amend` never amend commits that have been pushed to remote repositories.
+    2.  For `git rebase` never rebase a repository that others may work off of.
+    3.  For `git reset` never reset commits that have been pushed to remote repositories.
+    4.  For `git push --force` only use it when appropriate, use it with caution, and preferably default to using `git push --force-with-lease`.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. Read the chapter on [Branches covered by git-scm](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell) if you want an even deeper dive into branches.
+2. Read [think-like-a-git](http://think-like-a-git.net/)
+    *   Take your time with this resource as well, it's very well written and will be very helpful in solidifying your understanding of Git.
+
+</div>
+
+### Knowledge check
+
+This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
+
+- [A KNOWLEDGE CHECK QUESTION](A-KNOWLEDGE-CHECK-URL)
+
+### Additional resources
+
+This section contains helpful links to related content. It isn’t required, so consider it supplemental.
+
+* If you're looking for a fun way to deepen your knowledge of working with branches and Git, check out the popular game, [Learn Git Branching](https://learngitbranching.js.org/)
```

Comment:
```suggestion
* If you're looking for an interactive way to deepen your knowledge of working with Git, check out this game, [Learn Git Branching](https://learngitbranching.js.org/)
```
We don't really know how popular it is, and while I think it's a good tool I don't think many would find it "fun" in the typical sense.

Also the "branches and" seems unneeded since branches are part of Git.

The `the` -> `this` change just makes it read more naturally to me, but if that's grammatically incorrect just ignore this part, I'm no subject matter expert on grammar.